visualization/app/codeCharta/util/codeMapHelper.ts

```diff
@@ -2,38 +2,39 @@ import { hierarchy } from "d3-hierarchy"
 import { BlacklistItem, BlacklistType, CodeMapNode, MarkedPackage } from "../codeCharta.model"
 import ignore from "ignore"
 
-function getAnyCodeMapNodeFromPath(path: string, root: CodeMapNode): CodeMapNode {
+function getAnyCodeMapNodeFromPath(path: string, root: CodeMapNode) {
 	const matchingNode = hierarchy<CodeMapNode>(root)
 		.descendants()
-		.find(node => node.data.path === path)
-	return matchingNode ? matchingNode.data : null
+		.find(({ data }) => data.path === path)
+	return matchingNode?.data
 }
 
-function getCodeMapNodeFromPath(path: string, nodeType: string, root: CodeMapNode): CodeMapNode {
+function getCodeMapNodeFromPath(path: string, nodeType: string, root: CodeMapNode) {
 	const matchingNode = hierarchy<CodeMapNode>(root)
 		.descendants()
-		.find(node => node.data.path === path && node.data.type === nodeType)
-	return matchingNode ? matchingNode.data : null
+		.find(({ data }) => data.path === path && data.type === nodeType)
+	return matchingNode?.data
 }
 
-function getAllPaths(node: CodeMapNode): Array<string> {
+function getAllPaths(node: CodeMapNode) {
 	return hierarchy<CodeMapNode>(node)
 		.descendants()
-		.map(node => node.data.path)
+		.map(({ data }) => data.path)
 }
 
-function transformPath(toTransform: string): string {
-	let removeNumberOfCharactersFromStart = 0
+function transformPath(toTransform: string) {
+	let removeNumberOfCharactersFromStart = 2
 
-	if (toTransform.startsWith("./")) {
-		removeNumberOfCharactersFromStart = 2
-	} else if (toTransform.startsWith("/")) {
+	if (toTransform.startsWith("/")) {
 		removeNumberOfCharactersFromStart = 1
+	} else if (!toTransform.startsWith("./")) {
+		return toTransform
```

Comment:
Prevents slicing

Reply:
Before:
`transformPath("./root/path")` => `root/path`

after:
`transformPath("./root/path")` => `./root/path`

Reply:
This works as before, it just does less work :-)